app/controllers/contests_controller.rb

```diff
@@ -146,19 +162,49 @@ def destroy
   end
 
   def register
+    @teams = current_user.teams
+    @registration = @contest.find_registration(current_user)
+    team = @registration&.team
+    if team.present?
+      @teammate_registrations = team.users.map{|user|
```

Comment:
nit: will something like
```
@teammate_registrations = @contest.contest_registrations
                                  .where(team: team)
                                  .includes(:user)
                                  .map { |reg| [reg.user, reg] }
```
be better? not an RoR expert, but guess something like this could reduce the number of queries.